src/lib/routes/client-api/feature.ts

```diff
@@ -118,7 +128,22 @@ export default class FeatureController extends Controller {
                 featureQuery,
             );
         }
-        res.json({ version, features, query: featureQuery });
+
+        const response = {
+            version,
+            features,
+            query: featureQuery,
+        };
+
+        if (this.useGlobalSegments) {
```

Comment:
I think we also need to consider the memoizee for segments. I see a huge risk of segments and feature toggle configuration becoming out of sync in the cache period.

Reply:
Excellent point! I've included segmentation in the memoization call, these are two separate db queries so there's a very small risk of them being out of sync, but that's handled in the SDKs by forcing it to false in that case and there's client spec tests to ensure that that contract is maintained